### ISSUE TITLE
Keda prometheus scaler based on series query instead of api

### ISSUE THREAD

**Issue by yuvalweber** [kedacore/keda#3907]
the prometheus scaler uses a scale based on a normal query to "/api/v1/query"
https://github.com/kedacore/keda/blob/dab87b52d5ad56aca439ad9f5683e7ad641a4c60/pkg/scalers/prometheus_scaler.go#L224
I was wondering if there is a possibility to add the option to ask for a series metrics in "/api/v1/series?match[]={series}&start={current_timestamp}" once in like 5 intervals of the query (or something that the user will choose, under the trigger section), and then if the metric does not exist maybe change the value saved for the metric to something like "-1".
Now I know that in order to ask for series query we can't use functions or signs like ">" , "=" and more.
But I wrote an extractor that knows how to extract only the relevant series query from all of this (also we can ask the user to provide the series query for this existence check).
Based on This we could maybe create an alert that if the `keda_metrics_adapter_scaler_metrics_value` will be equal -1 then the metric the user issued does not return any relevant info cause or the object does not exist or it was deleted some how.

## Examples
1) let say I want to create a scaling based on the nginx requests my ingress accepts.
I write the query `sum(nginx_ingress_controller_requests{ingress="test1-ingress"})` and I accidentally write a typo, instead of "test-ingress" I wrote "test1-ingress" ingress which does not exist.
If I add the possibility in the trigger to ask for series query once in 5 intervals, I could see that this typo was made cause it will change the metric value to "-1" which does not exist.

2) another example relevant to this is the option that someone might removed this ingress and forgot to remove the scaledObject attached to it. Now it keeps asking prometheus each interval for this query although it's not relevant anymore.
If I add the series query I could determine this (because of the "-1" value) and remove it to reduce the load on prometheus

## example scaledObject 
```
apiVersion: keda.k8s.io/v1alpha1
kind: ScaledObject
metadata:
  name: prometheus-scaledobject
  namespace: default
spec:
  scaleTargetRef:
    deploymentName: my-deployment
  triggers:
  - type: prometheus
    metadata:
      serverAddress: http://prometheus:9090
      metricName: http_requests_total
      threshold: '100'
      query: sum(nginx_ingress_controller_requests{ingress="test1-ingress"})
      existence: # check every 10 minutes if the query still relevant (still return info)
          interval: 600 
```
or
```
apiVersion: keda.k8s.io/v1alpha1
kind: ScaledObject
metadata:
  name: prometheus-scaledobject
  namespace: default
spec:
  scaleTargetRef:
    deploymentName: my-deployment
  triggers:
  - type: prometheus
    metadata:
      serverAddress: http://prometheus:9090
      metricName: http_requests_total
      threshold: '100'
      query: sum(nginx_ingress_controller_requests{ingress="test1-ingress"})
      series: # check every 10 minutes if the query still relevant (still return info)
          interval: 600 
          query: nginx_ingress_controller_requests{ingress="test1-ingress"} # this will be issues as series query for existence check
```

**Comment by JorTurFer** [comment]
hello @yuvalweber ,
Sorry, I don't get the reason behind this request, could you rephrase it?

**Comment by yuvalweber** [comment]
hey @JorTurFer 
I updated my issue.
Let me know if it's more clear right now

**Comment by JorTurFer** [comment]
Okey, now it's clear ❤️ 
You'd like to have a mechanism to detect when the query isn't working. I'm not totally sure about if we are already covering the scenario you need, I mean, you could ensure that `ignoreNullValues` is `false` (default value is `true`). This changes the behavior when `null` is received from transforming it into 0 to return an error.
Returning error from scaler, you can check the `keda_metrics_adapter_scaled_object_error_totals` (in the operator) metric to detect it.
Could this match with your requirement?

**Comment by yuvalweber** [comment]
Yeah it actually matches my requirements.
I forgot that those metrics don't work like alerts that you write queries like: `sum(http_request_total) > 3`.
In our queries there should always be value that return something so `ignoreNullValues ` really good for me.
Thanks for the help @JorTurFer 
I close this issue

**Comment by JorTurFer** [comment]
Happy to help :)
